components/script/dom/urlsearchparams.rs

```diff
@@ -163,3 +164,23 @@ impl URLSearchParams {
         }
     }
 }
+
+
+impl Iterable for URLSearchParams {
+    type Key = USVString;
+    type Value = USVString;
+
+    fn get_iterable_length(&self) -> u32 {
+        self.list.borrow().iter().count() as u32
```

Comment:
Yes, it should, r=me with that. A test with something like what is described in #10351 would be great too :)

```
let a=new URL("http://a.b/c?a=1&b=2&c=3&d=4");
let b = a.searchParams;
for (i of b) {
    a.search="x=1&y=2&z=3";
    console.log(i);
}
// outputs
Array [ "a", "1" ]
Array [ "y", "2" ]
Array [ "z", "3" ]
```